src/catch2/internal/catch_template_test_registry.hpp

```diff
@@ -84,11 +84,12 @@
             struct TestName{\
                 TestName(){\
                     size_t index = 0;                                    \
-                    constexpr char const* tmpl_types[] = {CATCH_REC_LIST(INTERNAL_CATCH_STRINGIZE_WITHOUT_PARENS, __VA_ARGS__)};\
-                    using expander = size_t[];\
+                    constexpr char const* tmpl_types[] = {CATCH_REC_LIST(INTERNAL_CATCH_STRINGIZE_WITHOUT_PARENS, __VA_ARGS__)}; /* NOLINT(cppcoreguidelines-avoid-c-arrays,modernize-avoid-c-arrays) */\
+                    using expander = size_t[]; /* NOLINT(cppcoreguidelines-avoid-c-arrays,modernize-avoid-c-arrays) */\
                     (void)expander{(reg_test(Types{}, Catch::NameAndTags{ Name " - " + std::string(tmpl_types[index]), Tags } ), index++)... };/* NOLINT */ \
                 }\
             };\
+            /* NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables) */\
             static int INTERNAL_CATCH_UNIQUE_NAME( globalRegistrar ) = [](){\
```

Comment:
Actually, is there a reason not to make this `const`?

Reply:
I am still pretty new to the code. I did try with `const` and all the test seem to pass. I am assuming it might be ok to just make it `const`.

Reply:
Done: 5dab39f6

Reply:
If the tests compile & pass, it should be fine. This isn't a place that tests can't reach, so you can rely on them.